github_copilot_seats/seats.py

```diff
@@ -0,0 +1,113 @@
+import json
+import os
+from datetime import datetime, timezone, timedelta
+from typing import Any
+import autokitteh
+from autokitteh.github import github_client
+from autokitteh.slack import slack_client
+from helpers import github_username_to_slack_user_id
+
+GITHUB_ORG = os.getenv("GITHUB_ORG")
+IDLE_USAGE_THRESHOLD = int(os.getenv("IDLE_USAGE_THRESHOLD"))
+LOGINS = os.getenv("LOGINS")
+LOG_CHANNEL = os.getenv("LOG_CHANNEL")
+
+logins = LOGINS.split(",") if LOGINS else None
+github = github_client("github_conn")
+slack = slack_client("slack_conn")
+
+org = github.get_organization(GITHUB_ORG)
+copilot = org.get_copilot()
+
+
+def prune_idle_seats() -> list[dict]:
+    """Prunes idle GitHub Copilot users based on their last activity time."""
+    seats = find_idle_seats()
+    for seat in seats:
+        autokitteh.start(loc="seats.py:engage_seat", data=seat)
+    return seats
+
+
+def find_idle_seats() -> list[dict]:
+    """Identifies idle GitHub Copilot users based on their last activity time."""
+    seats = copilot.get_seats()
+    t = datetime.now(timezone.utc)
+    idle_seats = []
+
+    for seat in seats:
+        if logins and seat.assignee.login not in logins:
+            print(f"skipping {seat.assignee.login}")
+            continue
+
+        delta = t - seat.last_activity_at
+        is_idle = delta >= timedelta(minutes=IDLE_USAGE_THRESHOLD)
+
+        print(
+            f"{seat.assignee.login}: {t} - {seat.last_activity_at} = {delta} "
+            f"{'>=' if is_idle else '<'} {IDLE_USAGE_THRESHOLD} minutes"
+        )
```

Comment:
It's more readable in this case to use helper vars to avoid multi-line wrapping of a single `print` statement